src/ucp/api/ucp.h

```diff
@@ -939,20 +947,36 @@ typedef struct ucp_generic_dt_ops {
  * @ingroup UCP_DATATYPE
  * @brief UCP datatype attributes
  *
- * This structure provides attributes that can be queried for a UCP datatype.
+ * This structure provides attributes of a UCP datatype.
  */
 typedef struct ucp_datatype_attr {
     /**
      * Mask of valid fields in this structure, using bits from @ref
      * ucp_datatype_attr_field. Fields not specified in this mask will be
      * ignored. Provides ABI compatibility with respect to adding new fields.
      */
-    uint64_t field_mask;
+    uint64_t   field_mask;
+
+    /**
+     * Packed size of the given datatype. (output parameter)
+     */
+    size_t     packed_size;
 
     /**
-     * Packed size of a single element of the given datatype.
+     * Pointer to a data buffer of the associated data type.
+     * This field is optional.
+     * If @ref UCP_DATATYPE_ATTR_FIELD_BUFFER is not set in @ref field_mask,
+     * this field defaults to @e NULL.
      */
-    size_t   packed_size;
+    const void *buffer;
+
+    /**
+     * Number of elements in @a buffer.
+     * This value is optional.
+     * If @ref UCP_DATATYPE_ATTR_FIELD_COUNT is not set in @ref field_mask, the
+     * value of this field defaults to 1.
```

Comment:
would 0 be a better default size for a NULL buffer?

Reply:
Agree with @tonycurtis comment - 1 is odd

Reply:
We've had this topic before (https://github.com/openucx/ucx/pull/8150#discussion_r951272653). Intuitively, I would also have taken 0, but there are also arguments for 1. I do not have a strong opinion on this.

Reply:
@tonycurtis @shamisp if the default is 0, it means that if this parameter is not supplied the function will always return 0. and that is essentially as if we're making this parameter mandatory.
IMO it should be 1 by default since the neutral value w.r.t. multiplication is 1.

Reply:
`If @ref UCP_DATATYPE_ATTR_FIELD_COUNT is not set in @ref field_mask, the function executes the query for a single ucp_datatype_t` . I think this sounds better. @tonycurtis  ?

Reply:
@yosefe , @tonycurtis approved the above and I think it sounds better while does not change the meaning